### ISSUE TITLE
Test case failed on chain_test

### ISSUE THREAD

**Issue by HarukaMa** [bitshares/bitshares-core#1326]
**Instructions**
I've been testing building with Boost 1.67.0, and the compilation succeeded despite it is not supported, so I decided to run tests (`*_test`) to check for errors.

When running `chain_test`, I've met one error:

```
unknown location:0: fatal error: in "uia_tests/asset_name_test": std::runtime_error: collate_byname<char>::collate_byname failed to construct for en_US.UTF8
/Users/MrX/src/bitshares-core/tests/tests/uia_tests.cpp:503: last checkpoint
```

other tests except `es_test` (didn't have es installation locally) succeeded without errors.

Additional info:

```
$ locale
LANG="ja_JP.UTF-8"
LC_COLLATE="ja_JP.UTF-8"
LC_CTYPE="ja_JP.UTF-8"
LC_MESSAGES="ja_JP.UTF-8"
LC_MONETARY="ja_JP.UTF-8"
LC_NUMERIC="ja_JP.UTF-8"
LC_TIME="ja_JP.UTF-8"
LC_ALL=
```

**Impacts**
Describe which portion(s) of BitShares Core may be impacted by this bug. Please tick at least one box.
- [ ] API (the application programming interface)
- [ ] Build (the build process or something prior to compiled code)
- [ ] CLI (the command line wallet)
- [ ] Deployment (the deployment process after building such as Docker, Travis, etc.)
- [ ] DEX (the Decentralized EXchange, market engine, etc.)
- [ ] P2P (the peer-to-peer network for transaction/block propagation)
- [ ] Performance (system or user efficiency, etc.)
- [ ] Protocol (the blockchain logic, consensus, validation, etc.)
- [ ] Security (the security of system or user data, etc.)
- [ ] UX (the User Experience)
- [x] Other (please add below)

**Host Environment**
 - Host OS: macOS 10.13.6
 - Host Physical RAM: 16 GB
 - BitShares Version:  2.0.171025-minor-fix-1-1293-g0e86ff5 _NOTE: I think this string is broken for a while_
 - OpenSSL Version:  1.1.1-dev _it is using a self compiled version, I even forgot when I compiled it, but seems running fine_
 - Boost Version:   1.67.0

## CORE TEAM TASK LIST
- [ ] Evaluate / Prioritize Bug Report
- [ ] Refine User Stories / Requirements
- [ ] Define Test Cases
- [ ] Design / Develop Solution
- [ ] Perform QA/Testing
- [ ] Update Documentation

**Comment by abitmore** [comment]
The user doesn't have `en_US.UTF8` locale installed, while the test case is trying to use it. The failing test case is: https://github.com/bitshares/bitshares-core/blob/3e74426d301f098eb6306309b3dd80bb830a5c49/tests/tests/uia_tests.cpp#L505-L511

**Comment by abitmore** [comment]
Update: got same error in Windows.

**Comment by jmjatlanta** [comment]
I may be jumping in the middle here, but it is my understanding that this is testing to see if the isalpha function works as expected in various locales (new "non-C" way vs the classic "C" way).

What I do not understand is why are we testing a standard library function? Is there a belief that the standard will change? Or perhaps that some platforms will not implement it correctly?

Perhaps this test was simply to clarify how things work.

I may be wrong, but I do not see the benefit of those lines in this test. If everyone agrees, I will remove those lines. If not, please let me know why these lines are here. Thanks!

**Comment by pmconrad** [comment]
Resolved via #1346
